Detectors/FIT/FT0/simulation/src/Digits2Raw.cxx

```diff
@@ -124,35 +149,63 @@ void Digits2Raw::convertDigits(const o2::ft0::Digit& digit, const o2::ft0::LookU
           if ((nchannels % 2) == 1)
             mRawEventData.mEventData[nchannels] = {};
           mRawEventData.mEventHeader.nGBTWords = nGBTWords;
-          mPages[oldlink].write(mRawEventData.to_vector());
+          mPages[oldlink].write(mRawEventData.to_vector(0));
         }
         oldlink = nlink;
         mRawEventData.mEventHeader = makeGBTHeader(nlink, intRecord);
         nchannels = 0;
       }
       auto& newData = mRawEventData.mEventData[nchannels];
+      bool isAside = (d.ChId < 96);
       newData.charge = d.QTCAmpl;
       newData.time = d.CFDTime;
       newData.is1TimeLostEvent = 0;
       newData.is2TimeLostEvent = 0;
       newData.isADCinGate = 1;
       newData.isAmpHigh = 0;
       newData.isDoubleEvent = 0;
-      newData.isEventInTVDC = is0TVX ? 1 : 0;
+      newData.isEventInTVDC = digit.getisVrtx() ? 1 : 0;
       newData.isTimeInfoLate = 0;
       newData.isTimeInfoLost = 0;
       int chain = std::rand() % 2;
       newData.numberADC = chain ? 1 : 0;
       newData.channelID = lut.getMCP(d.ChId);
       LOG(DEBUG) << "packed GBT " << nlink << " channelID   " << (int)newData.channelID << " charge " << newData.charge << " time " << newData.time << " chain " << int(newData.numberADC) << " size " << sizeof(newData);
       nchannels++;
+      if (isAside) {
+        sumTimeA += d.CFDTime;
+        sumAmpA += d.QTCAmpl;
+        nChA++;
+      } else {
+        sumTimeC += d.CFDTime;
+        sumAmpC += d.QTCAmpl;
+        nChC++;
+      }
     }
     // fill mEventData[nchannels] with 0s to flag that this is a dummy data
     uint nGBTWords = uint((nchannels + 1) / 2);
     if ((nchannels % 2) == 1)
       mRawEventData.mEventData[nchannels] = {};
     mRawEventData.mEventHeader.nGBTWords = nGBTWords;
-    mPages[oldlink].write(mRawEventData.to_vector());
+    mPages[oldlink].write(mRawEventData.to_vector(0));
+    LOG(DEBUG) << " last " << oldlink;
+    mRawEventData.mEventHeader = makeGBTHeader(LinkTCM, intRecord); //TCM
+    mRawEventData.mEventHeader.nGBTWords = 1;
+    tcmdata.timeA = sumTimeA;
+    tcmdata.timeC = sumTimeC;
+    tcmdata.amplA = sumAmpA;
+    tcmdata.amplC = sumAmpC;
+    tcmdata.nChanA = nChA;
+    tcmdata.nChanC = nChC;
+    LOG(DEBUG) << "TCMdata"
+               << " time A " << int(tcmdata.timeA) << " time C " << int(tcmdata.timeC)
+               << " amp A " << int(tcmdata.amplA) << " amp C " << int(tcmdata.amplC)
+               << " N A " << int(tcmdata.nChanA) << " N C " << int(tcmdata.nChanC)
+               << " trig "
+               << " ver " << tcmdata.vertex << " A " << tcmdata.orA << " C " << tcmdata.orC
+               << " size " << sizeof(tcmdata);
+    mPages.at(LinkTCM).write(mRawEventData.to_vector(1));
+    LOG(DEBUG) << " write TCM " << LinkTCM;
```

Comment:
you also need to complete the last TF by writing empty pages for the HBFs of this (already started) TF, see ``testHBFUtils.cxx`` or the ITS code.